src/DbalCompatibility/ForeignKey.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\DbalCompatibility;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Schema\ForeignKeyConstraint;
+
+use function method_exists;
+
+/** @internal */
+final readonly class ForeignKey
+{
+    public function __construct(private ForeignKeyConstraint $foreignKey)
+    {
+    }
+
+    /** @return array<string> */
+    public function getReferencingColumns(AbstractPlatform $platform): array
+    {
+        if (! method_exists($this->foreignKey, 'getReferencingColumns')) {
+            return $this->foreignKey->getLocalColumns();
+        }
+
+        $namesAsStrings = [];
+
+        foreach ($this->foreignKey->getReferencingColumnNames() as $name) {
+            $namesAsStrings[] = $name->toSQL($platform);
```

Comment:
Should I maybe be using `toString` here? I fear that the platform won't be always available in places where I should use these methods, for instance here: https://github.com/doctrine/orm/blob/cf39e005537ad2ed13f43fe4edd9f5137e0390e6/src/Mapping/Driver/DatabaseDriver.php#L210

Reply:
If you want to represent names as strings for compatibility with DBAL 4, then `toString()` is fine. The contract is: `toSQL(AbstractPlatform)` will render the name so that it can be parsed by the platform in question; `toString()` will render the name so that it can be parsed by the corresponding name parser (i.e. converted back to object).